### DISCUSSION TITLE
Second DHCP request gets wrong address because it accepts response without a file

### DISCUSSION THREAD

**Discussion by max-allan** [ipxe/ipxe#1257] [Q&A] [answered]
Hi, I have 2 DHCP servers. One gives 192. addresses to anyone. The other gives 10. addresses to PXE clients
```
match if substring (option vendor-class-identifier, 0, 9) = "PXEClient";
```

Which is fine for PXE but for iPXE because if the offer doesn't include a file, it's ignored.

However, in iPXE the second request accepts any response with an IP address. So, it then fails.

There is a helpful example script:
```
 :retry
  ifconf -c dhcp && isset ${filename} || goto retry
```
But of course, I'm not even GETTING to the script stage... And it isn't clear if that will even make it accept other offers anyway.

Other than building my own undionly.kpxe that doesn't just accept any old DHCP offer, is there a way to "fix" this?

**Comment by mcb30** [comment, marked as answer]
> Other than building my own undionly.kpxe that doesn't just accept any old DHCP offer, is there a way to "fix" this?

Yes.  You need to fix your DHCP server setup so that you do not have two DHCP servers giving out conflicting information in the same physical network.

**Comment by NiKiZe** [comment]
As was already said, having 2 DHCP servers on the same network is invalid.
I can think of a few ways that you possibly could make this work, but it would still be a broken setup.
tcpdump or wireshark would be able to tell you what you need to know.

**Comment by max-allan** [comment]
IS this not what ProxyDHCP is all about? One server handles one group of clients and one handles everything else?

**Reply by NiKiZe** [reply]
ProxyDHCP does something completely different.
You have ONE server giving out ipaddresses, ProxyDHCP only gives out additional information such as next-server and filename, it is not giving out IP addresses.
